sql/core/src/test/scala/org/apache/spark/sql/MetadataCacheSuite.scala

```diff
@@ -57,13 +56,20 @@ abstract class MetadataCacheSuite extends QueryTest with SharedSQLContext {
         df.count()
       }
       assert(e.getMessage.contains("FileNotFoundException"))
-      assert(e.getMessage.contains("REFRESH"))
+      assert(e.getMessage.contains("recreating the Dataset/DataFrame involved"))
     }
   }
+}
+
+class MetadataCacheV1Suite extends MetadataCacheSuite {
+  override protected def sparkConf: SparkConf =
+    super
+      .sparkConf
+      .set(SQLConf.USE_V1_SOURCE_READER_LIST, "orc")
 
   test("SPARK-16337,SPARK-27504 temporary view refresh") {
```

Comment:
`SPARK-27504` had better be removed like [this](https://github.com/apache/spark/pull/24815/files#diff-0667b59236ca014a47b3fc20b6ea820eR41)?